### PR TITLE
Docs: add "Using Environment Variables".

### PR DESCRIPTION
**Pull request by concretevitamin**
Tested (run the relevant ones):

- [ ] Code formatting: `bash format.sh`
- [x] Any manual or new tests for this PR (please specify below): rendered locally
- [ ] All smoke tests: `pytest tests/test_smoke.py` 
- [ ] Relevant individual smoke tests: `pytest tests/test_smoke.py::test_fill_in_the_name` 
- [ ] Backward compatibility tests: `bash tests/backward_comaptibility_tests.sh`

### CODE REVIEW

File: docs/source/running-jobs/environment-variables.rst [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+
+Using in ``setup``
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+All user-specified environment variables are exported to a task's ``setup`` commands.
+
+Using in ``run``
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+All user-specified environment variables are exported to a task's execution (i.e., while its ``run`` commands are running).
```
Comment (Michaelvll) [inline docs/source/running-jobs/environment-variables.rst:50]: Maybe we can combine the two sections into one to reduce the duplication.
```
Using in ``setup`` and ``run``
```

File: docs/source/running-jobs/environment-variables.rst [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+       more :ref:`here <dist-jobs>`.
+     - 0
+   * - ``SKYPILOT_TASK_ID``
+     - A unique ID assigned to each task.
+       Useful for logging purposes: e.g., use a unique output path on the cluster; pass to Weights & Biases; etc.
+
+       If a task is run as a :ref:`managed spot job <spot-jobs>`, then all
+       recoveries of that job will have the same ID value. Read more :ref:`here <spot-jobs-end-to-end>`.
+     - sky-2023-07-06-21-18-31-563597_myclus_id-1
+
```
Comment (Michaelvll) [inline docs/source/running-jobs/environment-variables.rst:103]: There are a bunch more env vars added by the #2106, but it should be fine to leave it out, as that feature is still experimental.

**Review by Michaelvll — APPROVED** [review]
Thanks for adding this @concretevitamin! It is very important to keep track of the environment variables we have.

File: docs/source/running-jobs/environment-variables.rst (hunk 2 above)
Comment (concretevitamin) [reply docs/source/running-jobs/environment-variables.rst:100]: Sounds good.

File: docs/source/running-jobs/environment-variables.rst (hunk 1 above)
Comment (concretevitamin) [reply docs/source/running-jobs/environment-variables.rst]: Good call, done.

**Pull request merged**